src/community/security/oauth2-openid-connect/oauth2-openid-connect-core/src/main/java/org/geoserver/security/oauth2/ValidatingOAuth2RestTemplate.java

```diff
@@ -35,7 +36,9 @@ public ValidatingOAuth2RestTemplate(
             String jwkUri,
             OpenIdConnectFilterConfig config) {
         super(resource, context);
-        if (jwkUri != null) this.store = new JwkTokenStore(jwkUri);
+        if (jwkUri != null) {
+            this.store = new JwkTokenStore(jwkUri);
+        }
```

Comment:
Unnecessary change?

Reply:
This is actually the fix, the KID verification is optional